plasmapy/physics/tests/test_distribution.py

```diff
@@ -42,3 +47,84 @@ def test_Maxwellian_1D():
 
     with pytest.raises(ValueError):
         Maxwellian_1D(1*u.m/u.s, T=1*u.K, particle='XXX')
+
+def test_Maxwellian_speed_1D():
+    r"""test the 1D Maxwellian speed distribution function"""
+    T = 1.0 * u.eV
+    particle = 'H'
+    # get thermal velocity and thermal velocity squared
+    vTh = thermal_speed(T, particle=particle, method="most_probable")
+    # setting up integration from 0 to 10*vTh
+    xData1D = np.arange(0, 10.01, 0.01) * vTh
+    yData1D = Maxwellian_speed_1D(v=xData1D,T=T,particle=particle)
+    # integrating, this should be close to 1
+    integ = spint.trapz(y=yData1D, x=xData1D)
+    print("1D integral {0}".format(integ))
+    exceptStr = "Integral of distribution function should be 1."
+    assert np.isclose(integ.value, 1), exceptStr
+    
+    
+def test_Maxwellian_velocity_3D():
+    r"""test the 3D Maxwellian velocity distribution function"""
+    T = 1.0 * u.eV
+    particle = 'H'
+    # get thermal velocity and thermal velocity squared
+    vTh = thermal_speed(T, particle=particle, method="most_probable")
+    # defining a closure with partially applied arguments
+    def velDist3D(vx, vy, vz):
+        velDist = Maxwellian_velocity_3D(vx=vx,
+                                         vy=vy,
+                                         vz=vz,
+                                         T=T, 
+                                         particle=particle)
+        return velDist
+    # setting up integration from -10*vTh to 10*vTh, which is close to Inf
+    infApprox = 10 * vTh
+    # if I recall correctly, astropy.units may have trouble with
+    # accounting for units in integrals, so this step currently fails when
+    # tplquad tries to convert a quantity with units to a scalar.
+    # integrating, this should be close to 1
+    integ = spint.tplquad(velDist3D,
+                          -infApprox,
+                          infApprox,
+                          lambda z: -infApprox,
+                          lambda z: infApprox,
+                          lambda z, y: -infApprox,
+                          lambda z, y: infApprox,
+                          epsabs=1.49e-08*u.m/u.m, 
+                          epsrel=1.49e-08*u.m/u.m)
```

Comment:
I'm curious, why this number in particular? Also, the units seem off, could this perhaps be why `tplquad` failed here? If we're integrating velocities, don't we need `u.m/u.s`?

Reply:
Those are just the default values in tplquad 😉 , I put them in explicitly to check if adding dimensionless units would help the test behave correctly. Sometimes this works according to the "known issues" with astropy.units pages I linked below.

The final results from integration should be a probability, which is dimensionless, therefore the errors on the final value are dimensionless. `u.m/u.m` was just a shortcut for adding a dimensionless unit.

Here are some references for why the integration doesn't behave well with astropy.units. [Here](http://docs.astropy.org/en/stable/units/quantity.html#known-issues-with-conversion-to-numpy-arrays) and [here](http://docs.astropy.org/en/stable/known_issues.html#quantity-issues).

Reply:
Oh right, it's an error on the final value! I forgot about that, you're quite right.

Well, we can still go for floats on the internal side!